### PR TITLE
Add support for Scala 2.13; Fix warnings

### PR DESCRIPTION
**Pull request by cchantep**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,21 @@
-# Use Docker-based container (instead of OpenVZ)
-sudo: false
+language: scala
```
Comment (cchantep) [inline .travis.yml]: Update Travis CI configuration

File: .travis.yml [hunk 2]
```diff
@@ -1,27 +1,21 @@
...
     - $HOME/.sbt/boot
 
-language: scala
-
-# TODO - we'd like to actually test everything, but the process library has a deadlock right now
 jdk:
-  - oraclejdk8
-
-env:
-  matrix:
```
Comment (cchantep) [inline .travis.yml]: No need for matrix

File: build.sbt [hunk 3]
```diff
@@ -0,0 +1,145 @@
+import com.typesafe.tools.mima.core._, ProblemFilters._
```
Comment (cchantep) [inline build.sbt]: Refactor `project/build.scala` as `build.sbt`

File: build.sbt [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    Seq(
+      missingMethodInOld, // forward compat
+      ProblemFilters.exclude[IncompatibleTemplateDefProblem]("resource.AbstractManagedResource"), // previous change
+      ProblemFilters.exclude[MissingTypesProblem]("resource.DefaultManagedResource"), // previous change
+      ProblemFilters.exclude[UpdateForwarderBodyProblem]("resource.ManagedResourceOperations.toTraversable") // mixin for 2.13 compat
+    )
+  }
+)
+
+lazy val publishSettings: Seq[Setting[_]] = Seq(
```
Comment (cchantep) [inline build.sbt]: !! Without credentials, I cannot check the migration of the publish/site/jekyll settings

File: project/Compiler.scala [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import sbt.Keys._
+import sbt._
+import sbt.plugins.JvmPlugin
+
+object Compiler extends AutoPlugin {
```
Comment (cchantep) [inline project/Compiler.scala]: Enforce compiler options/flags

File: project/plugins.sbt [hunk 6]
```diff
@@ -2,13 +2,13 @@ resolvers += Resolver.url("sbt-plugin-releases", new URL("http://scalasbt.artifa
 
 resolvers += "jgit-repo" at "http://download.eclipse.org/jgit/maven"
 
-addSbtPlugin("com.typesafe.sbt" % "sbt-ghpages" % "0.5.2")
```
Comment (cchantep) [inline project/plugins.sbt]: Upgrade plugins

File: src/main/scala-2.13+/resource/Compat.scala [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package resource
+
+private[resource] object Compat {
```
Comment (cchantep) [inline src/main/scala-2.13+/resource/Compat.scala]: Scala 2.13 compat utility

File: src/main/scala-2.13-/resource/ManagedResourceCompat.scala [hunk 8]
```diff
@@ -0,0 +1,21 @@
+package resource
+
+import scala.collection.{ Traversable, TraversableOnce }
```
Comment (cchantep) [inline src/main/scala-2.13-/resource/ManagedResourceCompat.scala]: `Traversable` only for Scala 2.13-

File: src/main/scala/resource/Resource.scala [hunk 9]
```diff
@@ -6,14 +6,17 @@ import java.util.zip.GZIPOutputStream
...
  */
 trait Resource[R] {
   /**
-   * Opens a resource for manipulation.  Note:  If the resource is already open by definition of existence, then
+   * Opens a resource for manipulation.
+   * 
+   * *Note:* If the resource is already open by definition of existence, then
    * this method should perform a no-op (the default implementation).
    */
+  @com.github.ghik.silencer.silent
```
Comment (jsuereth) [inline src/main/scala/resource/Resource.scala]: Wasn't aware of this, what's it do?

**Comment by jsuereth** [comment]
Amazing cleanup!  Just one question, then good to merge and adapt.

There's probably a lot in here I'd like to cleanup if we're going to do a new release, but doing a quick 2.13 release would be great, I can work on that after this is merged.

File: src/main/scala/resource/Resource.scala (hunk 9 above)
Comment (cchantep) [reply src/main/scala/resource/Resource.scala]: Allow to ignore compiler warning only where not relevant (there the unused parameter)

**Pull request merged**